src/main/java/com/aws/greengrass/shadowmanager/ShadowManagerDatabase.java

```diff
@@ -57,21 +65,45 @@ public ShadowManagerDatabase(final Kernel kernel) {
      */
     public ShadowManagerDatabase(Path path) {
         this.dataSource = new JdbcDataSource();
-        this.dataSource.setURL(String.format(DATABASE_FORMAT, path));
+        this.dataSource.setURL(String.format(DATABASE_FORMAT, path, DATABASE_NAME));
+        this.databasePath = path.resolve(DATABASE_NAME + ".mv.db");
     }
 
     /**
      * Performs the database installation. This includes any migrations that needs to be performed.
      *
+     * @throws IOException     io exception
      * @throws FlywayException if an error occurs migrating the database.
      */
     @Synchronized
-    public void install() throws FlywayException {
+    public void install() throws IOException, FlywayException {
         Flyway flyway = Flyway.configure(getClass().getClassLoader())
                 .locations("db/migration")
                 .dataSource(dataSource)
                 .load();
-        flyway.migrate();
+        migrateDB(flyway);
+    }
+
+    private void migrateDB(Flyway flyway) throws IOException, FlywayException {
+        try {
+            flyway.migrate();
+        } catch (FlywaySqlException flywaySqlException) {
+            if (flywaySqlException.getCause() instanceof JdbcSQLNonTransientException) {
+                if (flywaySqlException.getCause().getCause() instanceof IllegalStateException) {
+                    logger.atWarn().cause(flywaySqlException).log("Shadow manager DB is corrupted. "
+                            + "Removing it and creating a new one.");
+                    recreateDB(flyway);
+                }
+            } else {
+                throw flywaySqlException;
+            }
+        }
+    }
+
+    private void recreateDB(Flyway flyway) throws IOException {
+        logger.atDebug().kv("database-path", databasePath.toString()).log("Deleting the existing DB");
+        Files.deleteIfExists(databasePath);
```

Comment:
how about the other database files? The database can have 3 or 4 files.

Reply:
We're setting up the database with `DEFRAG_ALWAYS=TRUE` which should ensure that only one db file exists. (Other files are tracelog and temp.db file). I'm deleting all the files in the workfolder that end with "db".